### ISSUE TITLE
should we introduce a ProxyError

### ISSUE THREAD

**Issue by rob-c** [ganga-devs/ganga#709]
Would people be objecting to introducing a ProxyError exception or even one called UserError?

This would allow us to raise exceptions which are defined as being due to a typo or mistake or bad code entered at the IPython prompt, rather than a bug in Ganga.

I'd like to make the prompt a little more friendly as some people associate the trace with something going horribly wrong rather than the trace being there to help them.
(I know a lot of people simply didn't care for the old logging information for instance)

Or, even should we disable the trace from a normal production release?

**Comment by milliams** [comment]
Isn't this the point of `GangaException`? Or am I misremembering?

**Comment by rob-c** [comment]
In this case which exceptions should we make stack traces for or not?
Is there merit in showing them to the user unless we've asked them to re-run with `--debug`?

**Comment by rob-c** [comment]
And in that case there are a few places we should be changing the exception type within the Proxy I suppose.

**Comment by egede** [comment]
The policy that we decided on a long time ago was that a `GangaException` should not show a stack trace but that others should .The idea behind this is that a `GangaException` is something that the user can expect to get back to them whereas any other exception in principle indicates that we have a problem with the code (and that the stack trace then can help with the bug report). Are we either not implementing this idea, or is there a wish to change what we do?

**Comment by rob-c** [comment]
In this case I'll review the exception handler but there are definitely some exceptions which could do without a colourful stacktrack. i.e.

the result of mistyping a command or attribute name on the prompt giving a trace is just annoying imo

**Comment by egede** [comment]
@rob-c Yes, I agree that we should try to catch `AttributeError` to avoid a thing like

```
Ganga In [6]: j.foo()
ERROR    Error: 'Job' object has no attribute 'foo'
ERROR    Unknown/Unexpected ERROR!!
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
/cvmfs/lhcb.cern.ch/lib/lhcb/GANGA/GANGA_v601r22/install/ganga/python/Ganga/GPI/__init__.py in <module>()
----> 1 j.foo()

/cvmfs/lhcb.cern.ch/lib/lhcb/GANGA/GANGA_v601r22/install/ganga/python/Ganga/GPIDev/Base/Proxy.py in _getattribute(self, name)
   1076                         returnable = addProxy(implInstance._attribute_filter__get__(name))
   1077             else:
-> 1078                 returnable = GPIProxyObject.__getattribute__(self, name)
   1079 
   1080         if isType(returnable, GangaObject):

AttributeError: 'Job' object has no attribute 'foo'
```

Others are OK though. If I try to assign an application to the backend, I get the more sensible

```
Ganga In [7]: j.backend=Executable()
ERROR    Error: GangaAttributeError:  ('backend: attempt to assign an incompatible object <Ganga.Lib.Executable.Executable.Executable object at 0x7f9fdc0b3950> to the property in category "backends found cat: applications"',)

[17:54:10]
Ganga In [8]: 
```

where at least we do not get the stacktrace. It would be good if we could avoid the useless reference `<Ganga.Lib.Executable.Executable.Executable object at 0x7f9fdc0b3950>` though.

**Comment by rob-c** [comment]
@edge I've addressed the first in #713 already I'll update the Proxy to have a more sensible `getName` call for the 2nd example you give which should help users.

I've also tried to catch all user commands which IPython can't execute. The trace offers no more information than simply displaying the error that the command is invalid imo.
